blog/fb4/index.qmd

```diff
@@ -0,0 +1,162 @@
+                                    orelse=[
+                                        Return(
+                                            value=Constant(value='Scalene'))])])])],
+            decorator_list=[])],
+    type_ignores=[])
+```
+
+A simple way to produce a valid mutated version of a program is through replacing some of its statements with ‘pass’. While MuFunctionAnalyzer utilizes mutations of specific functions, MuProgramAnalyzer is the main class responsible for mutation analysis of the test suite, accepting the name of the module to be tested and its source code. The method ‘getitem’ accepts the test module, fixes the import entries on the test module to correctly point to the mutant module, and passes it to the test runner ‘MutantTestRunner’. The problem lies with equivalent mutants, mutants indistinguishable from the original in terms of semantics, whereas it becomes very difficult to judge the mutation score in their presence. A solution to this is inspecting mutants manually if they’re small enough, or randomly selecting a smaller amount of mutants to manually inspect if there’s a larger scale of them. Chao’s estimator can also be utilized to compute the result of the complete test matrix of each test against each mutant.
+
+## Reflection
+
+The Fuzzing Book has a chapter that talks about something called mutation analysis. This is a way to check how good our tests are when we run software. Just because we test a lot doesn't mean the software will always work right. The book gives two examples, named ineffective_test_1 and ineffective_test_2, to show this point.
+
+In the same chapter, there's also a bit about a triangle program. This program figures out the type of triangle using the lengths of its three sides. To check how good the tests are for this program, they first make the program more standard. They do this by using a method called 'mutation'. After that, they change some parts of the program, like swapping 'return' with 'pass'. If the program works correctly, it should show an error because some triangle types won't be recognized.
+
+The chapter also mentions problems when we purposely add fake errors to test the program. This reminds us to be careful and make sure our tests are both thorough and able to find real mistakes in the software.
+
+
+## Use Cases
+
+Through utilization of mutations, we can more efficiently test how our existing test suites within the Chasten project will handle possible bugs. We can create a specific test case that can be used to collect sample inputs and methods, processing identified inventory, then trace the grammar and implemented code in order to run tests and generate inputs that can easily identify and eliminate any existing errors within Chasten. Mutation analysis as opposed to code coverage is a great way for us to have full confidence that Chasten will work on any and all possible inputs we may encounter, and allow us to pinpoint where exactly our code needs corrections.
```

Comment:
What is the "identified inventory"?

Reply:
I removed the phrase "identified inventory" because it was a bit confusing wording. I instead explained the methods and inputs that can be collected through utilization of mutations.